### ISSUE TITLE
Portainer [Error: WireGuard exited with the error: Cannot find device "wg0"]

### ISSUE THREAD

**Issue by rossiscool123** [WeeJeWel/wg-easy#415]
**Describe the bug**
A manual install via docker-compose returns no errors at all but trying to deploy with portainer gives the following error "Error: WireGuard exited with the error: Cannot find device "wg0""

**To Reproduce**
1. Deploy a container on portainer using the following image "https://hub.docker.com/r/weejewel/wg-easy"
2. Edit the environment variable "WG_HOST" to the servers IP address.
3. Start the container and check the logs

**Expected behavior**
It to actually work like a manual install.
**Screenshots**
![image](https://user-images.githubusercontent.com/62076535/201566466-cd225fd2-af67-4849-9c5f-c41f1dfe0c63.png)

**Additional context**
Server is a cloud instance provided by Hetzner
Portainer version is 2.16.0
Server is running Ubuntu 20.04.5 LTS

**Comment by eandersons** [comment]
Take a look at the `docker run` command from the [instructions](/WeeJeWel/wg-easy#2-run-wireguard-easy):

```
$ docker run -d \
  --name=wg-easy \
  -e WG_HOST=🚨YOUR_SERVER_IP \
  -e PASSWORD=🚨YOUR_ADMIN_PASSWORD \
  -v ~/.wg-easy:/etc/wireguard \
  -p 51820:51820/udp \
  -p 51821:51821/tcp \
  --cap-add=NET_ADMIN \
  --cap-add=SYS_MODULE \
  --sysctl="net.ipv4.conf.all.src_valid_mark=1" \
  --sysctl="net.ipv4.ip_forward=1" \
  --restart unless-stopped \
  weejewel/wg-easy
```

You have to specify all options in Portainer as well.

The most relevant part is this:

```
  --cap-add=NET_ADMIN \
  --cap-add=SYS_MODULE \
  --sysctl="net.ipv4.conf.all.src_valid_mark=1" \
  --sysctl="net.ipv4.ip_forward=1" \
```

`NET_ADMIN` and `SYS_MODULE` can be set in container creation page section's "Advanced container settings" tab "Capabilities". Enabling those capabilities was enough to successfully create the WireGuard Easy container without errors.

`sysctl` options can be specified in container creation page section's "Advanced container settings" tab "Runtime & Resources".

**Comment by skillspark** [comment]
@rossiscool123 Did you ever get this to work? 
I have the exact same issue, using Portainer 2.15.0 running on a Synology NAS (Linux-based OS). I installed the container using docker-compose.

**Comment by eandersons** [comment]
@skillspark, if you are using Docker Compose, then the Portainer part is not relevant and thus your problem is not the same as this issue.

Wrokarounds from https://github.com/WeeJeWel/wg-easy/issues/137#issuecomment-1069634947 or https://github.com/WeeJeWel/wg-easy/issues/137#issuecomment-1347419464 might be relevant in your case as their authors are also using Synology.

**Comment by rossiscool123** [comment]
Yeah making a Stack worked as intended. That's the correct way of running it.
